ADR0005: Decide on python code style guide

lukpueh opened the pull request:
Fixes #1128 

**Description of the changes being introduced by the pull request**:
Document why we chose to henceforth use the Google Python style guide with minimal refinements. 

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
ADR0005: Decide on python code style guide

Use Google style guide with refinements, because the Google style
guide is a comprehensive, well-established style guide that is
mostly based on PEP-8 and was accepted by everyone on the TUF team.

There is no need to replicate these recommendations. However, we do
provide a very slim document with additional refinements, in order
to emphasize on items the we consider especially important, want to
be handled differently, or in one specific way, where the Google
guide would allow multiple.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>
Co-authored-by: Joshua Lock <jlock@vmware.com>

joshuagl commented on docs/adr/0005-use-google-python-style-guide.md:
```diff
@@ -0,0 +1,38 @@
+
+* Flaws in original Secure Systems Lab style guide
+* Curating a complete custom style guide is time consuming
+* Well-established style rules lower contribution barrier
```
Nit: You might argue this is already covered by the "lower contribution barrier" driver, but this was a major factor for me:

```suggestion
* Well-established style rules lower contribution barrier
* Custom style is not supported by default in common tooling (i.e. editors
  and linters)
```

joshuagl commented on docs/adr/0005-use-google-python-style-guide.md:
```diff
@@ -0,0 +1,38 @@
+style guide is a comprehensive, well-established style guide that is mostly
+based on PEP-8 and was accepted by everyone on the TUF team. There is no need
+to replicate these recommendations. However, we do provide a very slim document
+with additional refinements, in order to emphasize on items the we consider
```
Nit: English language construct.
```suggestion
with additional refinements, in order to emphasize items the we consider
```

joshuagl reviewed (approved):
This looks good, thanks @lukpueh. I pointed out a couple of minor nits, but I'm OK with the ADR as is.

(Apologies for being nit-picky about English.)

trishankatdatadog commented:
Good idea. Could we also use pylint, flake8, and black?

lukpueh commented:
Thanks for the comments, you two! No need to apologize,  @joshuagl, I do appreciate language corrections, and our [style guide actually encourages](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#386-punctuation-spelling-and-grammar) them.

Commit:
Add style guide usage instructions to ADR0005

Similar instructions are in the style guide preamble, but we repeat
it here for emphasis.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented:
@joshuagl, I squashed your suggestions into my commit. @trishankatdatadog, I added another commit to mention linters and formatters and general style guide usage/transition instructions. Would either of you mind taking another look? Thanks!

trishankatdatadog reviewed (approved):
💯

trishankatdatadog commented:
Hmm, I do have some reservations regarding hanging vs aligned indents in function arguments, but whatever, I can live with it.

lukpueh commented:
> Hmm, I do have some reservations regarding hanging vs aligned indents in function arguments, but whatever, I can live with it.

What do you mean? Our [new guide specifically favors aligned indent](https://github.com/secure-systems-lab/code-style-guidelines/blob/master/python.md#indentation-and-line-continuation), which is what you seemed [very passionate above](https://github.com/theupdateframework/tuf/issues/1128#issuecomment-720371661).

trishankatdatadog commented:
> What do you mean? Our [new guide specifically favors aligned indent](https://github.com/secure-systems-lab/code-style-guidelines/blob/master/python.md#indentation-and-line-continuation), which is what you seemed [very passionate above](https://github.com/theupdateframework/tuf/issues/1128#issuecomment-720371661).

My bad, I missed the tweak to the Google guide. Keep calm and carry on 💯 💯
